lib/_http_client.js

```diff
@@ -22,6 +22,9 @@ function ClientRequest(options, cb) {
 
   if (typeof options === 'string') {
     options = url.parse(options);
+    if (!options.hostname) {
+      throw new Error('Unable to determine the domain name');
+    }
   } else {
     options = util._extend({}, options);
```

Comment:
@Fishrock123 If the user doesn't specify any `hostname` in the `options`, then defaulting to `localhost` is expected. But when I give `google.com` (without `http://`), it should not take me to `localhost`.